tests/access_control/test_auto_2fa.py

```diff
@@ -1,118 +1,95 @@
+            self.browser.get(self.sp_shib_url(sp))
+            self.log_in_netid(self.browser, netid3, assert_success=False)
+            self.enter_duo_passcode(self.browser)
 
     def test_auto_2fa_b(self, netid3):
         """
-        AC-1 Part B. Prompted for 2FA only on diafine7.
+        AC-1 Part B
         """
-        fresh_browser = Chrome()
+        # b. Prompted for 2FA only on diafine7.
         sp = ServiceProviderInstance.diafine6
         with self.utils.using_test_sp(sp):
-            fresh_browser.get(f'{self.sp_url(sp)}/shib{self.test_env}')
-            self.login_submit_form(fresh_browser, netid3, self.password)
-            fresh_browser.wait_for_tag('h2', f'{self.sp_domain(sp)} sign-in success!')
+            self.browser.get(self.sp_shib_url(sp))
+            self.log_in_netid(self.browser, netid3, match_service_provider=sp)
 
         sp = ServiceProviderInstance.diafine7
         with self.utils.using_test_sp(sp):
-            fresh_browser.get(f'{self.sp_url(sp)}/shib{self.test_env}')
-            self.two_fa_submit_form(fresh_browser)
-            fresh_browser.wait_for_tag('h2', f'{self.sp_domain(sp)} sign-in success!')
+            self.browser.get(self.sp_shib_url(sp))
+            self.enter_duo_passcode(self.browser)
+            self.browser.wait_for_tag('h2', f'{self.sp_domain(sp)} sign-in success!')
 
     def test_auto_2fa_c(self, netid3):
         """
-        AC-1 Part C. Prompted for pwd then 2FA on diafine7.
+        AC-1 Part C
         """
-        #
-        fresh_browser = Chrome()
+        # Prompted for pwd then 2FA on diafine7.
         sp = ServiceProviderInstance.diafine6
         with self.utils.using_test_sp(sp):
-            fresh_browser.get(f'{self.sp_url(sp)}/shib{self.test_env}')
-            self.login_submit_form(fresh_browser, netid3, self.password)
-            fresh_browser.wait_for_tag('h2', f'{self.sp_domain(sp)} sign-in success!')
+            with self.browser.tab_context():
+                self.browser.get(self.sp_shib_url(sp))
+                self.log_in_netid(self.browser, netid3, match_service_provider=sp)
 
         sp = ServiceProviderInstance.diafine7
         with self.utils.using_test_sp(sp):
-            fresh_browser.get(f'{self.sp_url(sp)}/shib{self.test_env}force')
-            self.login_submit_form(fresh_browser, netid3, self.password)
-            self.two_fa_submit_form(fresh_browser)
-            fresh_browser.wait_for_tag('h2', f'{self.sp_domain(sp)} sign-in success!')
-            fresh_browser.close()
+            with self.browser.tab_context():
+                self.browser.get(self.sp_shib_url(sp, append='force'))
+                self.log_in_netid(self.browser, netid3, match_service_provider=sp, assert_success=False)
+                self.enter_duo_passcode(self.browser)
 
     def test_auto_2fa_d(self, netid3):
         """
-        AC-1 Part D. No prompts on diafine7.
+        AC-1 Part D
         """
-        fresh_browser = Chrome()
```

Comment:
What is the replacement for the new browser instance that was getting created on line 73?

Reply:
`self.browser` is always a `fresh_browser`, if you look at the base class.

Reply:
I see it now, cool.